programs/merkle-distributor/src/state/merkle_distributor.rs

```diff
@@ -58,12 +71,66 @@ pub struct AirdropBonus {
     /// total bonus
     pub total_bonus: u64,
     // vesting slot duration
-    pub vesting_slot_duration: u64,
+    pub vesting_duration: u64,
     /// total bonus
     pub total_claimed_bonus: u64,
 }
 
+pub struct ActivationHandler {
+    /// current slot or current timestamp
+    pub curr_time: u64,
+    /// activation slot or activation timestamp
+    pub activation_time: u64,
+    /// bonus multiplier
+    pub airdrop_bonus: AirdropBonus,
+}
+
+impl ActivationHandler {
+    pub fn validate_claim(&self) -> Result<()> {
+        require!(
+            self.activation_time <= self.curr_time,
+            ErrorCode::ClaimingIsNotStarted
+        );
+        Ok(())
+    }
+    pub fn get_bonus_for_a_claimaint(&self, max_bonus: u64) -> Result<u64> {
+        let curr_time = self.curr_time;
+        let start_time = self.activation_time;
+        let end_time = self.airdrop_bonus.vesting_duration.safe_add(start_time)?;
+
+        if curr_time >= start_time {
+            if curr_time >= end_time {
+                Ok(max_bonus)
+            } else {
+                let slot_into_unlock = curr_time.safe_sub(start_time)?;
+                let total_unlock_slot = self.airdrop_bonus.vesting_duration;
```

Comment:
What about `total_unlock_duration` as the variable name?

Reply:
fixed